apps/website/src/pages/notifications/[notificationId].tsx

```diff
@@ -0,0 +1,106 @@
+import type {
+  GetStaticPropsContext,
+  InferGetStaticPropsType,
+  NextPage,
+} from "next";
+
+import React, { useEffect } from "react";
+
+import { useRouter } from "next/router";
+import Section from "@/components/content/Section";
+import Heading from "@/components/content/Heading";
+import Meta from "@/components/content/Meta";
+
+import { getNotificationById } from "@/server/db/notifications";
+import Link from "@/components/content/Link";
+
+export async function getStaticProps({ params }: GetStaticPropsContext) {
+  const notificationId = String(params?.notificationId || "");
+  if (notificationId === "") return { notFound: true };
+
+  const notification = await getNotificationById(notificationId);
+  if (!notification) return { notFound: true };
+
+  return {
+    props: {
+      notification,
+    },
+  };
+}
+
+export async function getStaticPaths() {
+  return {
+    paths: [], // Do we want to prerender some posts?
+    fallback: "blocking",
+  };
+}
+
+const NotificationPage: NextPage<
+  InferGetStaticPropsType<typeof getStaticProps>
+> = ({ notification }) => {
+  const router = useRouter();
+  useEffect(() => {
+    if (notification.linkUrl) {
+      window.location.href = notification.linkUrl;
+    } else {
+      router
+        .replace(
+          `/updates?${new URLSearchParams({ notificationId: notification.id })}`
+        )
+        .then(() => {
+          // ignore
+        });
+    }
+  });
+
+  const content = (
+    <>
+      <div className="text-center">
+        <Heading>{notification.title || "Update"}</Heading>
+        <p>{notification.message}</p>
+
+        {notification.linkUrl ? (
+          <Link external href={notification.linkUrl}>
+            {notification.linkUrl}
+          </Link>
+        ) : (
+          <Link
+            href={`/updates?${new URLSearchParams({
+              notificationId: notification.id,
+            })}`}
+          >
+            Show updates!
+          </Link>
+        )}
+      </div>
+    </>
+  );
+
+  return (
+    <>
+      <Meta
+        title={notification.title || "Update"}
+        description={notification.message}
+      />
```

Comment:
At some point it would be nice to use the uploaded notification image as `og:image`.